front/temporal/hard_delete/activities.ts

```diff
@@ -0,0 +1,118 @@
+import { Context } from "@temporalio/activity";
+import { QueryTypes, Sequelize } from "sequelize";
+
+import config from "@app/lib/production_checks/config";
+import logger from "@app/logger/logger";
+import type {
+  RunExecutionRow,
+  RunsJoinsRow,
+} from "@app/temporal/hard_delete/types";
+import {
+  getRunExecutionsDeletionCutoffDate,
+  isSequelizeForeignKeyConstraintError,
+} from "@app/temporal/hard_delete/utils";
+
+const BATCH_SIZE = 100;
+
+export async function purgeExpiredRunExecutions() {
+  const primaryCoreDatabaseUri = config.getPrimaryCoreDatabaseUri();
+  const coreSequelize = new Sequelize(primaryCoreDatabaseUri, {
+    logging: false,
+  });
+
+  const cutoffDate = getRunExecutionsDeletionCutoffDate();
+
+  logger.info(
+    {},
+    `About to purge block and run executions anterior to ${new Date(
+      cutoffDate
+    ).toISOString()}.`
+  );
+
+  let hasMoreRunsToPurge = true;
+  let runsPurgedCount = 0;
+  do {
+    const batchToDelete = await coreSequelize.query<RunExecutionRow>(
+      "SELECT id FROM runs WHERE created < :cutoffDate ORDER BY created, id ASC LIMIT :batchSize",
+      {
+        replacements: {
+          batchSize: BATCH_SIZE,
+          cutoffDate,
```

Comment:
I verified that node getTime() is in same epoch format (millis) than core created

Reply:
I tested the flow E2E on my laptop and it did work as expected.